bin/DataDownloadAndMerge.pm

```diff
@@ -95,8 +95,11 @@ sub paired_end_data {
             my $output_file =
                 $args_dict->{"final_output_file"} . "."
               . $args_dict->{"output_format"};
+            my $tmp_dir = tempdir( "readtools.DDR.XXXXX",
```

Comment:
For Picard, while downloading, Ram's code was using the following construction: [`my $picard_temp = "$args_dict->{'output_directory'}/$args_dict->{'random_id'}/".$mapper_name."_picard_tmp"`](https://github.com/robmaz/distmap/blob/de799f8938b79cb12b2d4b67335de9170ac6b4f7/bin/DataMerge.pm#L431)

We should do the same here because it is basically the same param in ReadTools, it will be consistent with the rest of the code, is mapper-aware and run-specific. I recommend:

```perl
my $tmp_dir = "$args_dict->{'output_directory'}/$args_dict->{'random_id'}/".$mapper_name."_readtools_tmp"
```

Reply:
Only that $args_dict->{'random_id'} is not actually a random ID and we'd need to clean and remove the folder explicitly to make sure that no old files enter into new merges.  I think the File::Temp core module is the official and safe way to do this.

Reply:
ReadTools is taking care of the files that are download: it checks the HDFS for them, overriding existing ones, and naming the files by itself. There is no risk of including old files into the downloaded data...